week_1/Array/product_of_array_except_self/product_of_array_except_self.js

```diff
@@ -0,0 +1,33 @@
+// Given an integer array nums, return an array answer such that answer[i] is equal to the product of all the elements of nums except nums[i].
+
+// The algorithm must run in O(n) time and can't use the division operator.
+/**
+ * @param {number[]} an array of numbers
+ * @return {number[]} an array of products
+ */
+
+const productExceptSelf = (nums) => {
```

Comment:
I've read this a couple times through and struggle to grasp the meaning of everything happening here. I'll have to circle back to this next time.

Reply:
Is there anywhere in particular I could try to clarify with a comment or a change in the variable names?

Reply:
One thing you could try is deriving the left and right multipliers in the same loop, then loop again to combine them

```js
let right = nums.length -1;
let left = 0;
while (left < nums.length) {
 ... find your multipliers
 right--;
 left++
}
```

Reply:
I went ahead and wrote the loop as you suggested, as well as rewriting the comments. It's a bit tough to balance clarity with brevity in an algorithm like this, so let me know how it looks.